test/intl402/PluralRules/constructor-option-read-order.js

```diff
@@ -0,0 +1,53 @@
+/// Copyright 2023 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-initializepluralrules
+description: Checks the order of option read.
+features: [Intl.NumberFormat-v3]
+includes: [compareArray.js]
+---*/
+
+let optionKeys = [
+    // Inside InitializePluralRules
+    "localeMatcher",
+    "type",
+    // Inside SetNumberFormatDigitOptions
+        "minimumIntegerDigits",
+        "minimumFractionDigits",
+        "maximumFractionDigits",
+        "minimumSignificantDigits",
+        "maximumSignificantDigits",
+        "roundingPriority",
+        "roundingIncrement",
+        "roundingMode",
+        "trailingZeroDisplay",
+    // End of SetNumberFormatDigitOptions
+];
+let expected = [
+    "localeMatcher",
+    "type",
+    "minimumIntegerDigits",
+    "minimumFractionDigits",
+    "maximumFractionDigits",
+    "minimumSignificantDigits",
+    "maximumSignificantDigits",
+    "roundingIncrement",
+    "roundingMode",
+    "roundingPriority",
+    "trailingZeroDisplay"
+];
+let readKeys = new Array();
+// For each item returned by resolvedOptions of default, add a getter
+// to track the reading order.
+let opt = {};
+optionKeys.forEach((property) =>
+    Object.defineProperty(opt, property, {
+        get() {
+            readKeys[readKeys.length] = property;
+            return undefined;
+        },
+    }));
+let p = new Intl.PluralRules(undefined, opt);
+assert.compareArray(readKeys, expected,
+    "GetOption should be called in the correct order");
```

Comment:
cf. https://github.com/tc39/test262/pull/3911#discussion_r1325084523
```suggestion

// Use getters to track the order of reading known properties.
// TODO: Should we use a Proxy to detect *unexpected* property reads?
let reads = new Array();
let options = {};
optionKeys.forEach((key) => {
    Object.defineProperty(options, key, {
        get() {
            reads.push(key);
            return undefined;
        },
    });
});
new Intl.PluralRules(undefined, options);
assert.compareArray(reads, optionKeys, "Intl.PluralRules options read order");
```